internal/datastore/mongo_store.go

```diff
@@ -0,0 +1,97 @@
+package datastore
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/SimplQ/simplQ-golang/internal/models"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// A Structure with Collections frequently used and a pointer to the client
+type MongoDB struct {
+    Client *mongo.Client
+    Queue *mongo.Collection
+    Token *mongo.Collection
+}
+
+var mongodb MongoDB
+
+func ConnectMongoDB(uri string) *MongoDB {
+    log.Println("Connection to MongoDB...")
+
+    client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+    
+    if err != nil {
+        log.Fatal(err)
+    }
+
+    mongodb = MongoDB {
+        client,
+        client.Database("simplq").Collection("queue"),
+        client.Database("simplq").Collection("token"),
+    }
+
+    log.Println("Successfully connected to MongoDB!")
+
+    return &mongodb
+}
+
+func (mongodb MongoDB) CreateQueue(queue models.Queue) models.QueueId {
+    // Only take required data from argument
+    // Ignore other values of the argument queue
+    queue_insert := bson.D{
+        // Get queue name from argument
+        {"queueName", queue.QueueName},
+        // not paused and not deleted initially
+        {"isPaused", false},
+        {"isDeleted", false},
+        // Current time is creation time
+        {"creationTime", time.Now()},
+        // Empty array of tokens
+        {"tokens", primitive.A{}},
+    }
```

Comment:
Let us take the queue object that we get as argument, and then put it as in.

No business logic should come here. This file is just mongo code:Code to store the passed queue into MongoDB.

Reply:
Tomorrow we might store queue object in portgres, and we would write an implementation for postgres. We don't want to copy the logic there.

Reply:
Makes sense I'll move this to the endpoint instead
But I think we will still have to remove `Id` from it so that MongoDB generates an ID on its own.

Reply:
I've moved the initialization to the create queue handler
But, I've assigned `queue.Id = ""` to make sure that the Id is generated by Mongo